lib/api/CDataFrameTrainBoostedTreeRunner.cc

```diff
@@ -173,43 +177,53 @@ CDataFrameTrainBoostedTreeRunner::CDataFrameTrainBoostedTreeRunner(
         m_CustomProcessors.CopyFrom(*parameters[FEATURE_PROCESSORS].jsonObject(),
                                     m_CustomProcessors.GetAllocator());
     }
-    if (alpha != -1.0 && alpha < 0.0) {
+
+    if (std::any_of(alpha.begin(), alpha.end(), [](double x) { return x < 0.0; })) {
```

Comment:
This would be a good candidate for a function to avoid redundancy and improve readability.

Reply:
I thought about this but many of the conditions are slightly different so I'd have to have different named functions for each possible condition we have on the variable. In the end I decided it was best to just go with the lambda approach.